anyway/infographics_utils.py

```diff
@@ -1118,22 +1120,53 @@ def localize_items(request_params: RequestParams, items: Dict) -> Dict:
+#  for matching with the "involved_markers_hebrew" table (see SQLAlchemy below). Maybe we should make InjuredAccidentsWithPedestriansWidget
+# inherit Widget instead of SubUrbanWidget?
 class InjuredAccidentsWithPedestriansWidget(SubUrbanWidget):
     name: str = "injured_accidents_with_pedestrians"
 
+    def validate_parameters(self):
+        # TODO: validate each parameter and display message accordingly
+        return self.request_params.location_info.get('yishuv_name') is not None and \
+            (self.request_params.news_flash_obj.street1_hebrew is not None or   \
+            self.request_params.news_flash_obj.street2_hebrew is not None) and  \
+            self.request_params.years_ago is None
+
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, type(self).name)
+
+        if not self.validate_parameters():
+            logging.exception(f"Could not validate parameters for {NewsFlash} : {request_params.news_flash_obj.id}")
+            return None
+
         self.rank = 18
-        self.text = {"title": "נפגעים הולכי רגל ברחוב ז׳בוטינסקי, פתח תקווה"}
+        self.text = {"title": f"נפגעים הולכי רגל ברחוב(ות) {str.join(', ', [request_params.news_flash_obj.street1_hebrew, request_params.news_flash_obj.street2_hebrew])}, {request_params.news_flash_obj.yishuv_name}"}
 
     @staticmethod
+    # TODO: change?
     def is_in_cache() -> bool:
         return False
 
     def generate_items(self) -> None:
-        self.items = (
-            InjuredAccidentsWithPedestriansWidget.injured_accidents_with_pedestrians_mock_data()
-        )
+        try:
+            query = db.session.query(InvolvedMarkerView)\
+                .with_entities(InvolvedMarkerView.accident_year,    \
+                    InvolvedMarkerView.injury_severity, \
+                    InvolvedMarkerView.injury_severity_hebrew,\
+                    func.count(distinct(f"{InvolvedMarkerView.provider_and_id}_{InvolvedMarkerView.involve_id}")))  \
+                .filter(InvolvedMarkerView.accident_yishuv_name == self.request_params.location_info['yishuv_name'])\
+                .filter(InvolvedMarkerView.injured_type == 1)   \
+                .filter(or_(InvolvedMarkerView.street1_hebrew == self.request_params.news_flash_obj.street1_hebrew,
+                    InvolvedMarkerView.street2_hebrew == self.request_params.news_flash_obj.street2_hebrew))    \
+                .filter(InvolvedMarkerView.accident_year >= datetime.date.today().year - self.request_params.years_ago)   \
+                .group_by(InvolvedMarkerView.accident_year, InvolvedMarkerView.injury_severity, InvolvedMarkerView.injury_severity_hebrew)
+
+            self.items = (query.all())
+        except Exception as e:
+            logging.exception(f"InjuredAccidentsWithPedestriansWidget.generate_items(): {e}")
+            return None
```

Comment:
Should we just return None on exception? Not sure it's the best way to handle it.
@ziv17 what do you think?

Reply:
I think not catching the exception is better as it allows us to decide centrally whether to fail the whole request or just give up on this widget and display the rest

Reply:
from my little experience debugging the app locally, most of the time you cannot get any indication of what is wrong. perhaps it should be part of a wider effort, to standardize and formalize how we deal with exceptions and more importantly, how we log them for debugging purposes

Reply:
I agree to some extent, still I think it is better to write a detailed error message `logging.error()`, and re-raise the exception in this case.

Reply:
What ever you both decide here is ok with me.

Reply:
will re-raise and log